articles/supervised-learning-algorithms/index.md

```diff
@@ -0,0 +1,154 @@
+
+   - Linear Kernel
+   - Polynomial Kernel
+   - Radial Basis Function Kernel
+
+Let us look at implementing SVM using `sklearn`.
+
+   ```python
+    # Import necessary modules
+    from sklearn import svm
+    from sklearn.model_selection import train_test_split
+    from sklearn.datasets import load_digits
+    # Create feature and target arrays
+    digits = load_digits()
+    X = digits.data
+    y = digits.target
+
+    # Split into training and test set
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=42, stratify=y)
+
+    # Create a SVM classifier
+    classifier = svm.SVC(kernel='poly')
+    classifier.fit(X_train, y_train)
+    print(classifier.score(X_test,y_test))
+   ```
+
+### Regression
+Regression algorithms are another subset of machine learning algorithms used to predict continuous numeric responses. As seen in an earlier example, predicting house rent given different factors is an example of regression. Let's look at the regression algorithm and use ridge regression as an example.
+
+3. Linear Regression: Linear regression is a simple yet effective method used in a large number of applications. Let us say we have an input feature vector `x`. The output feature vector `y` is the predicted entity. We use the sum of least squares to compute the relation between the target and input variables. Linear regression can be implemented using sklearn. Let us look at the implementation below:
+
+    ```python
+    # Import necessary modules
+    from sklearn.linear_model import LinearRegression
+    from sklearn.model_selection import train_test_split
+    from sklearn.datasets import load_digits
+    import numpy as np
+    import matplotlib.pyplot as plt
+    # Create feature and target arrays
+
+    digits = load_digits()
+    X = digits.data
+    y = digits.target
+
+    # Split into training and test set
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=42, stratify=y)
+
+    linear_regression = LinearRegression()
+    linear_regression.fit(X_train,y_train)
+
+    print(linear_regression.score(X_test,y_test))
+    ```
+
+The output of the print statement will be 0.55. The scoring metric used for linear regression is the R^2 metric. Pronounced as R-squared, it tells us about the effectiveness of the curve fitting on the graph. The closer the value of R-squared is to 1, the higher chances of good curve fitting. Curve fitting may lead to overfitting when the number of features considered is less. Overfitting refers to a scenario where the model performs very well on the data it has seen. But its performance drops when it works on unseen data. Underfitting is also a possibility when we don't have sufficient data to train the model(s) with.
```

Comment:
This paragraph confused me a bit - when it came to the curve fitting - it may have been me - since Im not familiar with models - but the mention of a curve - made me pause for a moment and ask what curve are they talking about - it took me a second - to understand all these models are represented on graphs. 
The overfitting is kind of confusion - is there any way to include a bit more about that. A few sentences explaining the curve fitting -